classes/DepositPoint.inc.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * @file plugins/generic/sword/classes/DepositPoint.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2003-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class DepositPoint
+ * @ingroup plugins_generic_sword_classes
+ *
+ * @brief Deposit point entity.
+ */
+
+class DepositPoint extends DataObject {
+	/**
+	 * Get the ID of the deposit point.
+	 * @return int
+	 */
+	public function getId() {
+		return $this->getData('depositPointId');
+	}
+
+	/**
+	 * Set the ID of the deposit point
+	 * @param $feedId int
+	 */
+	public function setId($depositPointId) {
+		$this->setData('id', $depositPointId);
+		return $this->setData('depositPointId', $depositPointId);
+	}
+
+	/**
+	 * Get the context ID of the deposit point.
+	 * @return int
+	 */
+	public function getContextId() {
+		return $this->getData('contextId');
+	}
+	
+	/**
+	 * Set the journal ID of the deposit point.
+	 * @param $contextId int
+	 */
+	public function setContextId($contextId) {
+		return $this->setData('contextId', $contextId);
+	}
+
+	/**
+	 * Get deposit point display sequence.
+	 * @return float
+	 */
+	public function getSequence() {
+		return $this->getData('seq');
+	}
+
+	/**
+	 * Set deposit point display sequence
+	 * @param $seq float
+	 */
+	public function setSequence($seq) {
+		return $this->setData('seq', $seq);
+	}
+
+	/**
+	 * Get the localized name
+	 * @return string
+	 */
+	public function getLocalizedName() {
+		return $this->getLocalizedData('name');
+	}
+
+	/**
+	 * Get deposit point name
+	 * @param $locale string
+	 * @return string
+	 */
+	public function getName($locale) {
+		return $this->getData('name', $locale);
+	}
+
+	/**
+	 * Set deposit point name
+	 * @param $title string
+	 * @param $locale string
+	 */
+	public function setName($name, $locale) {
+		return $this->setData('name', $name, $locale);
+	}
+
+	/**
+	 * Get SWORD username
+	 * @return string
+	 */
+	public function getSwordUsername() {
+		return $this->getData('sword_username');
```

Comment:
Not a big deal, but for PHP-land indexes we tend to use `lowerCamelCase`.

Reply:
Alright. but in this case `sword_username` is the table column. Am I missing something?

Reply:
In PHP we use `lowerCamelCase` conventions; in SQL it's `underscore_separators`.

Reply:
`return $this->getData('sword_username');`
Sure. So this being a table column name is the above line correct or incorrect?

Reply:
The mapping from PHP (`lowerCamelCase`) to SQL (`underscore_separators`) happens in `_fromRow`:

     $obj->setData('someColumnName', $row['some_column_name']);

...and similar in `updateObject` and `insertObject`. So here, since it's PHP-land, it should be

    return $this->getData('swordUsername');

These conventions are somewhat arbitrary, but we're consistent with them.

Reply:
Okay, I see what you meant. updated.